cmd/metal-api/internal/service/machine-service.go

```diff
@@ -1957,6 +2027,15 @@ func (r machineResource) chassisIdentifyLEDOff(request *restful.Request, respons
 	r.machineCmd("chassisIdentifyLEDOff", metal.ChassisIdentifyLEDOffCmd, request, response, request.QueryParameter("description"))
 }
 
+func (r machineResource) updateFirmware(request *restful.Request, response *restful.Response) {
+	var p v1.MachineUpdateFirmware
+	err := request.ReadEntity(&p)
+	if checkError(request, response, utils.CurrentFuncName(), err) {
+		return
+	}
+	r.machineCmd("updateFirmware", metal.UpdateFirmwareCmd, request, response, p.Kind, p.Revision, p.Description, r.s3Client.Url, r.s3Client.Key, r.s3Client.Secret)
```

Comment:
Would it make sense to check if the firmware exists before handing it down to the metal-core?